contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -1366,6 +1536,144 @@ public void processQueue() {
+				ltiProps.put("lis_person_name_full", ufn + " " + uln);
+				ltiProps.put("lis_person_name_given", ufn);
+				ltiProps.put("user_id", currentItem.getUserId());
+				
+				//TODO move, repeated
+				ContentResource resource = null;
+				ResourceProperties resourceProperties = null;
+				String fileName = null;
+				try {
+					try {
+						resource = contentHostingService.getResource(currentItem.getContentId());
+
+					} catch (IdUnusedException e4) {
+						//ToDo we should probably remove these from the Queue
+						log.warn("IdUnusedException: no resource with id " + currentItem.getContentId());
+						dao.delete(currentItem);
+						errors++;
+						continue;
+					}
+					resourceProperties = resource.getProperties();
+					fileName = resourceProperties.getProperty(resourceProperties.getNamePropDisplayName());
+					fileName = escapeFileName(fileName, resource.getId());
+				}
+				catch (PermissionException e2) {
+					log.error("Submission failed due to permission error.", e2);
+					currentItem.setStatus(ContentReviewItem.SUBMISSION_ERROR_NO_RETRY_CODE);
+					currentItem.setLastError("Permission exception: " + e2.getMessage());
+					dao.update(currentItem);
+					releaseLock(currentItem);
+					errors++;
+					continue;
+				}
+				catch (TypeException e) {
+					log.error("Submission failed due to content Type error.", e);
+					currentItem.setStatus(ContentReviewItem.SUBMISSION_ERROR_NO_RETRY_CODE);
+					currentItem.setLastError("Type Exception: " + e.getMessage());
+					dao.update(currentItem);
+					releaseLock(currentItem);
+					errors++;
+					continue;
+				}
+
+				//TII-97 filenames can't be longer than 200 chars
+				//TODO test
+				if (fileName != null && fileName.length() >=200 ) {
+					fileName = truncateFileName(fileName, 198);
+				}
+
+				//TODO check if getSubmissionId exists?
+				String[] parts = currentItem.getTaskId().split("/");
+				log.debug(parts[parts.length -1] + " " + parts.length);
+				String httpAccess = serverConfigurationService.getString("serverUrl") + "/access/assignment/s/" + currentItem.getSiteId() + "/" + parts[parts.length -1] + "/" + currentItem.getSubmissionId();
+				log.debug("httpAccess url: " + httpAccess);//debug
+				ltiProps.put("custom_submission_url", httpAccess);
+				ltiProps.put("custom_submission_title", fileName);
+				ltiProps.put("custom_submission_filename", fileName);
+				ltiProps.put("custom_xmlresponse","1");//mandatatory
+				//TODO params for callbacks y processing resubmission?
+				
+				String tiiId = null;
+				try {
+					org.sakaiproject.assignment.api.Assignment a = assignmentService.getAssignment(currentItem.getTaskId());
+					AssignmentContent ac = a.getContent();
+					ResourceProperties aProperties = ac.getProperties();
+					tiiId = aProperties.getProperty("turnitin_id");
+				} catch (IdUnusedException e) {
+					log.error("IdUnusedException: no assignment with id: " + currentItem.getTaskId());
```

Comment:
These duplicate catch blocks should use a utility method which removes most of the boilerplate code.